### PR TITLE
Remove dependancy on "six" and use OOP paradigm

### PR DESCRIPTION
**Pull request by SchnWalter**
And I ignored the 80 chars rules... it's more like ~95 in some cases. but I still tried to make lines 80 chars long without reducing the code readability.

### CODE REVIEW

**Comment by patcon** [comment]
Thanks @SchnWalter! For context before I review, can you summarize any functional changes (if any), or is it just a pure refactor?

(Also, I've been MIA myself, working on some other stuff, but haven't peaced out of this project -- just working on sorting out over-the-air ROM update infrastructure in my spare time, rather than the ROM itself :)

**Comment by SchnWalter** [comment]
Hello,
There are no functional changes, tried to keep any changes out, but somehow this small docs change slipped in: https://github.com/mission-impossible-android/mission-impossible-android/commit/d478e9ef17e9b35d846ffd66d135596e079beca -- Looks like bash actually ignores the new line when a backslash is present before the EOL, which has added to provide single line descriptions for the basic Zsh completion.

OTA updates, nice!

Good luck!

**Comment by patcon** [comment]
Coolio. Looks great! (I've just now realized commit-by-commit reviews are much less intimidating. Your commits are super-clear :)
